db/table_cache.cc

```diff
@@ -96,6 +96,7 @@ Status TableCache::GetTableReader(
   if (sequential_mode && ioptions_.compaction_readahead_size > 0) {
     file = NewReadaheadRandomAccessFile(std::move(file),
                                         ioptions_.compaction_readahead_size);
+    file->Hint(RandomAccessFile::SEQUENTIAL);
```

Comment:
This changes env_posix behavior too. Although I can't think of a use case where it causes problem but I would feel more comfortable if we don't change the existing behavior. How about go back to the previous solution of calling file->PrepareReadAhead() or something like that?